src/main/java/com/palantir/gradle/revapi/ConfigManager.java

```diff
@@ -29,6 +31,8 @@
     // application managing everything.
     private static final Object CONFIG_FILE_LOCK = new Object();
 
+    @InputFile
+    @OutputFile
```

Comment:
Don't think this file should be involved in any sort of caching.
Especially since the input is not correct when you run a task that nests this manager, the real input would only be correct by the time you enter the synchronized block in `modifyConfigFile`...